src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -204,5 +204,8 @@ static bool hasUnscopedRefAttribute(MethodSymbol? containingMethod)
                 }
             }
         }
+
+        internal sealed override bool UseUpdatedEscapeRules
+            => _containingMethod?.UseUpdatedEscapeRules ?? _containingType.ContainingModule.UseUpdatedEscapeRules;
```

Comment:
would it be simpler for all source symbols to just implement this as `ContainingModule.UseUpdatedEscapeRules`?

Reply:
This type is under `Symbols/Source/`, so I got the impression this is only used for source symbols, but maybe I am mistaken about that.

Reply:
`ThisParameterSymbol` is used for `PEMethodSymbol` and perhaps other cases as well.